source/Octopus.Tentacle/Communications/TentacleCommunicationsModule.cs

```diff
@@ -22,11 +23,19 @@ protected override void Load(ContainerBuilder builder)
             {
                 var configuration = c.Resolve<ITentacleConfiguration>();
                 var services = c.Resolve<IServiceFactory>();
-                var halibutRuntime = new HalibutRuntimeBuilder()
+                var halibutRuntimeBuilder = new HalibutRuntimeBuilder()
                     .WithServiceFactory(services)
                     .WithServerCertificate(configuration.TentacleCertificate)
-                    .WithMessageSerializer(serializerBuilder => serializerBuilder.WithLegacyContractSupport())
-                    .Build();
+                    .WithMessageSerializer(serializerBuilder => serializerBuilder.WithLegacyContractSupport());
+
+                //if we have the environment variable to enable async halibut, enable it :)
+                if (bool.TryParse(Environment.GetEnvironmentVariable(EnvironmentVariables.TentacleEnableAsyncHalibut), out var isEnabled) && isEnabled)
+                {
+                    halibutRuntimeBuilder.WithAsyncHalibutFeatureEnabled();
```

Comment:
🤔 I was just wondering whether we want to have a feature toggle at all, and just turn on Halibut async always.

We [have a story](https://app.shortcut.com/octopusdeploy/story/56576/make-tentacle-use-async-halibut) to do this work at some point.

But thinking it through, this is the safest way to unblock you, and when we get around to do that story, we can remove this environment variable then.

Reply:
Yeah, I wasn't sure what your plans are (outside of we are just turning it on), so I figured this was the safest then you can just trash that environment variable :)